### ISSUE TITLE
What Type of Hyperparameter Tuning Algorithm Does Ray Tune Use with YOLOv11?

### ISSUE THREAD

**Issue by bjh03205** [ultralytics/ultralytics#17989]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, thank you for your continuous efforts to develop better models.
I am using Ray Tune to tune the hyperparameters of YOLO11.

I understand that Ray Tune is widely recognized as a library of tuning algorithms that simplifies the process of hyperparameter tuning.

However, when using Ray Tune, I would like to know exactly which algorithm is being used to tune the hyperparameters.

Is there a way to check which configuration algorithm is running when using Ray Tune?

Additionally, if the default setting uses a genetic algorithm, what considerations should be made when determining the number of epochs?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @bjh03205, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a ❓ Question, as in this case, please provide as much information as possible about your setup, use case, and configurations to help us understand your requirements better. Specifically, details regarding which version of `ultralytics` and Ray Tune you are using, and any specific setup or logs would be very helpful.

In the meantime, make sure that you have consulted Ray Tune’s documentation to understand some of its common tuning algorithms like [Bayesian Optimization, Random Search, and Hyperband](https://docs.ray.io/en/latest/tune/api_docs/schedulers.html) to see if they align with your observations.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

While you're here, please ensure your `ultralytics` package is up-to-date to maximize compatibility with frameworks like Ray:

```bash
pip install -U ultralytics
```

## Upgrade

If you encounter unrelated issues while using YOLO, always verify that the latest version resolves them by upgrading your `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help point you in the right direction. An Ultralytics engineer will review your question and assist you further as soon as possible. Thanks for your patience! 😊

**Comment by Y-T-G** [comment]
It uses RandomSearch by default.

https://docs.ray.io/en/latest/tune/api/doc/ray.tune.TuneConfig.html
